internal/balancergroup/balancergroup.go

```diff
@@ -506,3 +506,13 @@ func (bg *BalancerGroup) ExitIdle() {
 	}
 	bg.outgoingMu.Unlock()
 }
+
+// ExitIdleOne instructs the sub-balancer `id` to exit IDLE state, if
+// appropriate and possible.
+func (bg *BalancerGroup) ExitIdleOne(id string) {
```

Comment:
Hmm, did I misunderstand anything?
How does this solve the race you found?
`exitIdle` still reads `scToSubBalancer` without holding incomingMu.

I was thinking adding a `func (bg *BalancerGroup) connectSCs(sbc *subBalancerWrapper)`, which holds incomingMu, and call `Connect()` on the subconns.

Reply:
This is not related to the race. I need this functionality anyways for RLS. I'm working on a PR for the race. The fix is easy, but I'm trying to see if I can get a testcase working for that scenario.